src/Compilers/CSharp/Portable/Binder/ForEachEnumeratorInfo.cs

```diff
@@ -21,13 +21,17 @@ internal sealed class ForEachEnumeratorInfo
         public readonly MethodSymbol CurrentPropertyGetter;
         public readonly MethodSymbol MoveNextMethod;
 
-        // Dispose method to be called on the enumerator (may be null).
+        // True if the enumerator needs disposal once used. 
+        // Will be either IDisposable/IAsyncDisposable, or use DisposeMethod below if set
         // Computed during initial binding so that we can expose it in the semantic model.
-        public readonly bool NeedsDisposeMethod;
+        public readonly bool NeedsDisposal;
 
         // When async and needs disposal, this stores the information to await the DisposeAsync() invocation
         public AwaitableInfo DisposeAwaitableInfo;
 
+        // When using pattern-based Dispose, this stores the method to invoke to Dispose
+        public MethodSymbol DisposeMethod;
```

Comment:
>public [](start = 8, length = 6)

`readonly`

Reply:
If this is for pattern only suggest putting that into the name to make call site usage clear.

---
In reply to: [250004490](https://github.com/dotnet/roslyn/pull/32640#discussion_r250004490) [](ancestors = 250004490)